python/ray/data/impl/pandas_block.py

```diff
@@ -164,6 +174,73 @@ def _empty_table() -> "pandas.DataFrame":
     def _sample(self, n_samples: int, key: "SortKeyT") -> "pandas.DataFrame":
         return self._table[[k[0] for k in key]].sample(n_samples, ignore_index=True)
 
+    def count(self, on: KeyFn, ignore_nulls: bool) -> Optional[U]:
+        if on is not None and not isinstance(on, str):
+            raise ValueError(
+                "on must be a string or None when aggregating on Pandas blocks, but "
+                f"got: {type(on)}."
+            )
+
+        col = self._table[on]
+        return col.count()
+
+    def sum(self, on: KeyFn, ignore_nulls: bool) -> Optional[U]:
+        if on is not None and not isinstance(on, str):
+            raise ValueError(
+                "on must be a string or None when aggregating on Pandas blocks, but "
+                f"got: {type(on)}."
+            )
+
+        col = self._table[on]
+        if col.isnull().all():
```

Comment:
This short-circuit is important for correctness. By default, Pandas will sum a column of all nulls to 0 if `skipna=True`, when what we actually want is `None` in that case. Since we can't disambiguate between a valid sum result of 0 or an all-null sum result of 0 after `_apply_agg()`, we need to do an explicit all-null check beforehand.